lib/practice_exercises.rb

```diff
@@ -1,13 +1,36 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+
+  reduced_string = string.downcase.gsub(/[^A-Za-z]/, "")
+
+  i = 0
+  while i < (reduced_string.length - 1)
+    if reduced_string[i] != reduced_string[reduced_string.length - i - 1]
+      return false
+    end
+    i += 1
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+# Time Complexity: O(n^2)
+# Space Complexity: O(n)
 def longest_prefix(strings)
```

Comment:
👍 However the time complexity is O(nm) where n is the number of strings and m is the length of the shortest string.